python/dgl/graphbolt/__init__.py

```diff
@@ -9,33 +9,36 @@
 pattern resulting in better memory usage for minibatch GNN training workloads.
 See https://pytorch.org/docs/stable/notes/cuda.html#optimizing-memory-usage-with-pytorch-cuda-alloc-conf,
 and set the environment variable `PYTORCH_CUDA_ALLOC_CONF=expandable_segments:False`
-if you want to disable it.
+if you want to disable it and set it True to acknowledge and disable the warning.
 """
 cuda_allocator_env = os.getenv("PYTORCH_CUDA_ALLOC_CONF")
-if cuda_allocator_env is None:
-    os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
-    gb_warning(CUDA_ALLOCATOR_ENV_WARNING_STR)
-else:
-    configs = {
+WARNING_STR_TO_BE_SHOWN = None
+configs = (
+    {}
+    if cuda_allocator_env is None
+    else {
         kv_pair.split(":")[0]: kv_pair.split(":")[1]
         for kv_pair in cuda_allocator_env.split(",")
     }
-    if "expandable_segments" in configs:
-        if configs["expandable_segments"] != "True":
-            gb_warning(
-                "You should consider `expandable_segments:True` in the"
-                " environment variable `PYTORCH_CUDA_ALLOC_CONF` for lower"
-                " memory usage. See "
-                "https://pytorch.org/docs/stable/notes/cuda.html"
-                "#optimizing-memory-usage-with-pytorch-cuda-alloc-conf"
-            )
-    else:
-        configs["expandable_segments"] = "True"
-        os.environ["PYTORCH_CUDA_ALLOC_CONF"] = ",".join(
-            [k + ":" + v for k, v in configs.items()]
+)
+if "expandable_segments" in configs:
+    if configs["expandable_segments"] != "True":
+        WARNING_STR_TO_BE_SHOWN = (
+            "You should consider `expandable_segments:True` in the"
+            " environment variable `PYTORCH_CUDA_ALLOC_CONF` for lower"
+            " memory usage. See "
+            "https://pytorch.org/docs/stable/notes/cuda.html"
+            "#optimizing-memory-usage-with-pytorch-cuda-alloc-conf"
         )
-        gb_warning(CUDA_ALLOCATOR_ENV_WARNING_STR)
-
+else:
+    configs["expandable_segments"] = "True"
```

Comment:
If Torch already made the feature on by default, do we need this code to double turn it on?

Reply:
I think it will be on by default only for 2.5. It helps with 2.4 and earlier tremendously as well. In the future, we can check version and display warnings only for older versions.